pkg/utils/types.go

```diff
@@ -14,20 +18,44 @@ type EngineDetails struct {
 	SvcAccount   string
 	AppKind      string
 	AppNamespace string
-	Config       *rest.Config
 }
 
 // ExperimentDetails is for collecting all the experiment-related details
 type ExperimentDetails struct {
-	Env       map[string]string
-	ExpLabels map[string]string
-	ExpImage  string
-	ExpArgs   []string
-	JobName   string
+	Name       string
+	Env        map[string]string
+	ExpLabels  map[string]string
+	ExpImage   string
+	ExpArgs    []string
+	JobName    string
+	Namespace  string
+	ConfigMaps []v1alpha1.ConfigMap
+	Secrets    []v1alpha1.Secret
+	VolumeOpts VolumeOpts
+	SvcAccount string
+}
+
+//VolumeOpts is a strcuture for all volume related operations
+type VolumeOpts struct {
+	VolumeMounts   []corev1.VolumeMount
+	VolumeBuilders []*volume.Builder
 }
 
 // ClientSets is a collection of clientSets needed
 type ClientSets struct {
 	KubeClient   *kubernetes.Clientset
 	LitmusClient *clientV1alpha1.Clientset
 }
+
+// GenerateClientSets will generation both ClientSets (k8s, and Litmus)
+func GenerateClientSets(config *rest.Config) (*kubernetes.Clientset, *clientV1alpha1.Clientset, error) {
```

Comment:
Move this function to `pkg/kubernetes/clientset.go` for kubernetes clinetset and `pkg/litmus/clientset.go` for litmus clientset